ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/CastorFullOptimization.java

```diff
@@ -648,20 +739,44 @@ private double computeOptimalSetpoint(double currentSetpoint, double cnecMargin,
             optimalSetpoint = minSetpointInAlignedRa;
         }
 
-        boolean oneRaIsNotPst = false;
         for (RangeAction<?> rangeAction : alignedRangeActions) {
             if (rangeAction instanceof PstRangeAction) {
-                if (oneRaIsNotPst) {
-                    throw new FaraoException(String.format("Pst range action %s is aligned with non Pst range actions.", rangeAction.getName()));
-                }
-                optimalSetpoint = ((PstRangeAction) rangeAction).convertTapToAngle(((PstRangeAction) rangeAction).roundUpAngleToTapWrtInitialSetpoint(optimalSetpoint, currentSetpoint));
-            } else {
-                oneRaIsNotPst = true;
+                optimalSetpoint = roundUpAngleToTapWrtInitialSetpoint((PstRangeAction) rangeAction, optimalSetpoint, currentSetpoint);
+                return optimalSetpoint;
             }
         }
         return optimalSetpoint;
     }
 
+    /**
+     * This function converts angleToBeRounded in the angle corresponding to the first tap
+     * after angleToBeRounded in the direction opposite of initialAngle.
+     */
+    public static Double roundUpAngleToTapWrtInitialSetpoint(PstRangeAction rangeAction, double angleToBeRounded, double initialAngle) {
+        rangeAction.checkAngle(initialAngle);
+        rangeAction.checkAngle(angleToBeRounded);
```

Comment:
do we really need this since min and max are already respected ?
plus what would happen if there is an exception thrown here ?